### ISSUE TITLE
Template was retained after changing location

### ISSUE THREAD

**Issue by jhen913** [likeastore/ngDialog#184]
Hi,

I am encountering this problem wherein the template used for the dialog was retained after closing and changing location. So I have:

index.html
login.html - contains the login page (path: /login, controller: LoginController)
cashier.html - (path: /cashier, controller: CashierController)
firstTemplate.html - template that contains the Logout button and other buttons

The scenario is:
a. [In /cashier] I will click a button that will trigger the dialog to open. This dialog contains the Logout button. This is how I declared the Logout button:

< a href="" ng-click="logout()">LOGOUT< /a >

b. The logout() method is declared in LoginController and accessible in $rootScope:

$rootScope.logout = function() {
..
ngDialog.close();
$location.path("/login");
}

c. After clicking Logout, the app is being directed in /login and shows the login page. However, the firstTemplate.html retains at the bottom of the page (but just a plain html and not like a dialog). I did an inspect element and verified that the firstTemplate was still there.

d. I played around with the code and removed the $location.path("/login"); line. The dialog closes and the firstTemplate is removed.

Can you check if I'm doing something wrong here? Or if it is really a bug. Btw, this is an AngularJS app. Thank you!

**Comment by egor-smirnov** [comment]
@jhen913 My guess would be that state change is done before dialog is closed - and dialog is closed not immediately. `close()` method of ngDialog returns promise that will be resolved after closing of the dialog.

Will it work if you use `closePromise`  documented at https://github.com/likeastore/ngDialog#closepromise-promise like below?

```javascript
$rootScope.logout = function() {
    ..
    ngDialog.close().then(function() {$location.path("/login");});
}
```

**Comment by jhen913** [comment]
Hello Egor,

I followed your suggestion, but element was still retained. When I inspect the page, the ngDialog element was still there. I noticed also that whenever a new ngDialog is created, it creates a new div outside my div with ng-view. 

So I have my index.html:

< div ng-view >< /div >

When I logout, it looks like this:

< div ng-view >< /div >
< div id="ngdialog2" class="ngdialog ngdialog-theme-default ng-scope ngdialog-closing" >
...
< /div >

Then when I login again, the logout dialog appears for awhile then disappears immediately as if that was the only time it was destroyed.
